### ISSUE TITLE
Add additional behavior to SimpleWindow so that it does not do a system.exit() when last window is closed.

### ISSUE THREAD

**Issue by ErikBjare** [lunduniversity/introprog#31]
Was discussed on the lunch meeting 2016-05-09.

The issue regards a change in behavior as following: Add additional behavior to SimpleWindow so that it does not do a system.exit() when last window is closed.

Should not affect existing code using the SimpleWindow class.

Proposed solution is to implement a setter that can be used to disable the behavior.

Added code:

    private boolean exitOnClose = true;   // The existing behavior
    public static void setExitOnClose(boolean exitOnClose) {
        this.exitOnClose = exitOnClose;
    }

Modifications to the event handler for the close event is also required.

Gustav has been assigned to give feedback and review the PR.

Sorry for the messy issue, it was rushed.

**Comment by bjornregnell** [comment]
Thanks, I'll implement it accordingly and test it after Gustav has had a look at your proposal.

**Comment by ErikBjare** [comment]
@bjornregnell Not sure if you missed it but there is already a PR ready to be tested: #32

**Comment by gustavcedersjo** [comment]
Looks good, but maybe it should be called something like `setExitOnLastClose` to describe when the exit occurs?

**Comment by bjornregnell** [comment]
Yeas that's a better name.
I guess the attribute should be:
 *static* private boolean exitOnLastClose = true;   
?

**Comment by bjornregnell** [comment]
Yes, good; but if I'm going to push this file later to another repo where the Java courses have this file in its older version, I don't think they want all the changes to line endings... I might just paste this code in and not mess with line endings ;)

**Comment by bjornregnell** [comment]
Fixed now. Have a look here and tell me if something looks strange:
https://github.com/lunduniversity/introprog/commit/fd7d52df1fb7cfb50e70e1a0b19d96ef6c9111a7

**Comment by ErikBjare** [comment]
@gustavcedersjo My PR already used that naming since I realized so aswell when implementing it :smile: 

@bjornregnell Nope, looks fine. One thing though, it is usually considered bad practice to reimplement a PR and close it without merging the commits in the PR. It's not a biggie in this instance (since the changes were minor and not a significant amount of work) but it can be quite a frustrating experience for the contributor if the changes in their PR are copied into the repo without attribution by including their commits. A better way is simply to tell the one making the PR to fix what is missing, this way you wont step on any toes and save yourself some time in the process :wink: 

Closing.
